read_tasks.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+todo_task_file=$1
+padding="       "
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+lines=$( wc -l < "$todo_task_file" )
+i=0
+echo ""
+
+while [[ $i -lt $((lines)) ]]; do
+  i=$((i+1))
+  task_line="$( sed "${i}q;d" "$todo_task_file" )"
+  task="$(echo $task_line| cut -d']' -f 2)"
+  task_status="$(echo $task_line| cut -d']' -f 1)"
+  
+  if [ "$task_status" = "[>" ]; then
+    # The selected task should be bold
+    printf "${bold}${padding}%s${normal}\n\n" "$task"  
```

Comment:
It's hard for me to notice the bold text, maybe it's due to the font I'm using. It would be nice to actually show an indicator like `> ` or `* `. This is how it looks when `You` are highlighted for me:

![image](https://user-images.githubusercontent.com/3248948/70391535-a4650100-19d6-11ea-9120-9623d9398840.png)

Reply:
How about ***-> bold italic*** is that too excessive??
<img width="379" alt="Screen Shot 2019-12-08 at 18 26 14" src="https://user-images.githubusercontent.com/2800451/70393231-3d504800-19e8-11ea-8413-cf85fa1742a4.png">

Reply:
Oh, setting another color seems like a good idea! Maybe force a color instead of letting the terminal determine it with it's default settings, like the TODO text? Also, did you add the `->` arrow because I don't see that one. :O

Reply:
Yes, I added the arrow. I'm currently working on fixing your comments. I wonder what color is good to use? I like the blue one on black background. Wonder how it looks on white though...

Reply:
Like this?
<img width="349" alt="Screen Shot 2019-12-08 at 18 46 42" src="https://user-images.githubusercontent.com/2800451/70393517-22330780-19eb-11ea-84af-84806427b40e.png">

Reply:
White looks horrible though, but its at least readable :p
<img width="353" alt="Screen Shot 2019-12-08 at 18 59 20" src="https://user-images.githubusercontent.com/2800451/70393693-edc04b00-19ec-11ea-8155-780869540453.png">

Reply:
Looks great!

Reply:
Not sure if intentional but I like the fact that this becase a background color in my terminal:

![image](https://user-images.githubusercontent.com/3248948/70393913-cc149300-19ef-11ea-8201-4b39b564d37f.png)

Reply:
Haha! Strange! Maybe a background color is better?

Reply:
What terminal do you use? Your colors are cooler than mine 🤩

Reply:
like this?
<img width="361" alt="Screen Shot 2019-12-08 at 19 37 04" src="https://user-images.githubusercontent.com/2800451/70394130-3595a100-19f2-11ea-8052-df578585f7ef.png">

Reply:
I'm using [iTerm2](https://iterm2.com/) with [Fira Code](https://github.com/tonsky/FiraCode).

I'm not sure how you determine colors, I'm not familiar with that way but it doesn't seem to be the default background color:
![image](https://user-images.githubusercontent.com/3248948/70394531-4a743380-19f6-11ea-9dce-cbd45cce5976.png)

I tend to use `\[\033[0m\]` to reset the default terminal background color. This is from my old bash config before I switched to zsh:

```bash
    local NONE="\[\033[0m\]"    # Unsets color to term's fg color

    # Regular colors
    local K="\[\033[0;30m\]"    # Black
    local R="\[\033[0;31m\]"    # Red
    local G="\[\033[0;32m\]"    # Green
    local Y="\[\033[0;33m\]"    # Yellow
    local B="\[\033[0;34m\]"    # Blue
    local M="\[\033[0;35m\]"    # Magenta
    local C="\[\033[0;36m\]"    # Cyan
    local W="\[\033[0;37m\]"    # White

    # Bolded colors
    local EMK="\[\033[1;30m\]"
    local EMR="\[\033[1;31m\]"
    local EMG="\[\033[1;32m\]"
    local EMY="\[\033[1;33m\]"
    local EMB="\[\033[1;34m\]"
    local EMM="\[\033[1;35m\]"
    local EMC="\[\033[1;36m\]"
    local EMW="\[\033[1;37m\]"

    # Background colors
    local BGK="\[\033[40m\]"
    local BGR="\[\033[41m\]"
    local BGG="\[\033[42m\]"
    local BGY="\[\033[43m\]"
    local BGB="\[\033[44m\]"
    local BGM="\[\033[45m\]"
    local BGC="\[\033[46m\]"
    local BGW="\[\033[47m\]"
```

Could you try to use `\[\033[46m\]` as background color and reset it with `\[\033[0m\]`. See how the `TODO` text is colored for reference.

Reply:
Oh, and [this](https://github.com/martinlindhe/base16-iterm2) is my color palette!

Reply:
I just googled setting bash background colors. But we get very different results. Ill try your suggesion :)

Reply:
Does this work for you now?

Reply:
Sadly, no :( 

![image](https://user-images.githubusercontent.com/3248948/70395615-ac399b00-1a00-11ea-859e-6d7233ee1a3b.png)

It seems like it's the italic thing that makes it work bad.  Without italic:

![image](https://user-images.githubusercontent.com/3248948/70395624-d12e0e00-1a00-11ea-9e3a-141766f6cf17.png)

And if I set the color manually to black:

![image](https://user-images.githubusercontent.com/3248948/70395633-f91d7180-1a00-11ea-9663-5ec7a3a2e15e.png)

The final image has this diff:

```diff
diff --git a/read_tasks.sh b/read_tasks.sh
index 34a27c1..118a942 100644
--- a/read_tasks.sh
+++ b/read_tasks.sh
@@ -1,7 +1,6 @@
 #!/usr/bin/env bash
 
-italic="\e[3m"
-normal=$(tput sgr0)
+color="\033[0;30m"
 background_color="\033[46m"
 reset_color="\033[0m"
 
@@ -14,10 +13,10 @@ while [[ $i -lt $((lines)) ]]; do
   task_line="$(sed "${i}q;d" "$todo_task_file")"
   task="$(echo "$task_line"| cut -d']' -f 2)"
   task_status="$(echo "$task_line"| cut -d']' -f 1)"
-  
+
   if [ "$task_status" = "[>" ]; then
     # The selected task should be bold
-    printf "${padding}${italic}${background_color} %s ${normal}${reset_color}\n\n" "$task"  
+    printf "${padding}${color}${background_color} %s ${reset_color}\n\n" "$task"
   else
     # All other tasks should be normal
     printf "${normal}${padding}%s\n\n" "$task"
```

Sadly I'm not too comfortable with how terminals render these things and what determines the exact output. [This](https://askubuntu.com/a/985386) post was quite helpful but sadly it doesn't combine color, background color and font style in one line.

Reply:
Let's remove the italic and also set the background color to black